### ISSUE TITLE
What is the recommended configuration for multiple-computer backup with safety?

### ISSUE THREAD

**Issue by zbasbuyuk** [gilbertchen/duplicacy#448]
Hi,
I'd like to find out if directory-names can be automatically stored (maybe even enforced)  in storage because I am mainly confused/surprised about repository name. Is there a case which user should use same repository name for backing up different directories? If there is none, can the repo name giving be automated to use something like UNC? 

Wishlist:
1) Backing up multiple computers to one storage for dedup.
Fear:
1) Configuration problems: having same repo name on multiple computers.
2) Curious/Malicious users playing with .duplicacy configuration files and causing problems.

So far I gathered that repository name is just that. Repository is just a name for the duplicacy to identify backup-sets. Maybe I am wrong here but my tests showed that if repo names are the same, the tool assumes they are modifications to the same data set. 
My experiment was as follows.
A1) create directory. 
A2) put  file  A in it. 
A3) initialize repo with name "test". 
A4) Backup repo.

B1) create another directory. 
B2) put file B ini it. 
B3) Initialize repo with "test". Software informs repository already initialized. 
B4) Backup repo.

B5) restore repo "test". Both files show up on the directory.

This means repo name is actually very critical if multiple data sources need to be dumped into the same storage location.

Since ".duplicacy " directory is in user's PCs, there is nothing stopping them messing with it.

What do people use for this type of scenario? What is the recommended safe method for using duplicacy in an office network?

Thank you,
Zeki

**Comment by TheBestPessimist** [comment]
You would be correct: `repository name` (i'm pretty sure in some places is also called `snapshot name`) is the unique identifier of a `backup set` (a folder which you are backing up).

I would suggest you name your backups in a format like `<pcname> - <folder to backup>`. Use a "-" and not a "/", "\"  for the name.


> Since ".duplicacy " directory is in user's PCs, there is nothing stopping them messing with it.

That's also true, but if said users want the backup to work, they shouldn't touch it.

> What do people use for this type of scenario? What is the recommended safe method for using duplicacy in an office network?

I'm just going to point out the (stupidly) obvious here: could you put a txt file in the repository folder with the name something along the lines of `PLEASE DO NOT MODIFY THIS FOLDER'S CONTENT EVER!`? So, basically, can't you ask people nicely to mind their own business and leave your files alone?

**Comment by opmetal** [comment]
If you're the admin of these users PCs you could use Windows group policy to set the privileges on this directory and prevent them from reading its contents.

**Comment by TheBestPessimist** [comment]
@opmetal: is that useful if the other people can go as admin as well? (dk how windows privileges work, but i got all sort of headache from "this file belongs to <someone else>" for files in my own user profile.

**Comment by zbasbuyuk** [comment]
I was thinking of a substitution-macro on the _repository_name_ which would simplify this approach. The macro substitution could allow user to append/prepend concatenation of fields such as  machine-name, IP address, MAC ADDRESS, directory, etc.

@TheBestPessimist  It seems the short answer to "SafeBackup Policy/Guideline for Networks" is TBD. 

Which is fine. The suggestions above are getting us closer. 

One issue I see is that, there are multiple locations on the PC what need backup and I think the resolution is via inclusion/exclusion.

My whole goal is to automate backups upto a point which I only get notified with exceptions along with a daily short summary. Something like 
**YYYY/MM/DD DONE  XXGBs  Backed up:  BackupSize increased by XX GBs
  13 PCs Succeeded: 2 PCs failed: 1 Machine AWOL ** 


If such a macro existed, one could kind-of build a big-boy's centralized backup system.

1) I could put duplicacy configuration files on root level of the drive.
2) I could protect directory with admin rights
3) All directories that need to be backed up, I would insert as inclusion/exclusion fields.

Then there would be few steps left to do.
a) automate backup-result reporting.
b) somehow retrieve backup times of each machine to a reporting tool.

I suspect duplicacy does not check if a machine is backed up or not. 

I could solve both a) and b) problems above with a few scripts.
1) create a directory with a common-admin repo name in every machine (location probably in duplicacy configuration folder).
2) create a file name that uniquely identifies each machine (macro above remember!) 
3) The file could potentially have machine's configuration parameters too just in case things get lost and need to build from scratch. 
4) The backup script could append start times, end times and log results in that file. 
5) on admin-machine one could "restore" directory to its last version (each file has a unique name anyway).
5) write a script to interpret logs and report the data above.


Zeki

**Comment by TheBestPessimist** [comment]
I don't think you wanna start from scratch, so here's something: https://github.com/gilbertchen/duplicacy/wiki/Scripts-and-utilities-index

**Comment by NiJoao** [comment]
I got a poor man's version of that going on:

1. My family computers all have a repo at "C:/Duplicacy/"
2. Simlink-mode to "C:/" (Which causes loop-bugs in old programs like SpaceMonger 1.4, but I can live without them)
3. Initialized via powershell as `.duplicacy\duplicacy_win_x64_2.1.0.exe init -e -c 4M $env:computername sftp://USER@SERVER/Duplicacy` to my 24h-ON desktop (Server), which most of the time is in the same network as them, but is port-forward and available via internet.
4. Global filter file for all possible useful file based on the [TheBestPessimist filter](https://github.com/TheBestPessimist/duplicacy-utils)
5. Backing up each 2 hours, with IDLE priority, 0x1 processor affinity, and limited to 1000KBps. Again, configured with the help of [TheBestPessimist utilities](https://github.com/TheBestPessimist/duplicacy-utils). No prunes nor checks on their PCs, its almost undetectable when running.
6. After each successful backup, each PC report with a ping to its https://healthchecks.io check. Which sends me an e-mail when a PC goes 7 days without backup.
7. The Server running the SFTP server (bitvise) is configured with a dummy repo to that common storage and to `duplicacy add -copy default -bit-identical -e offsite samba://NAS_ON_OTHER_SIDE_OF_HOUSE`
8. Daily, at night, the Server prunes both storage locations:
`duplicacy prune ...` and 
`duplicacy prune -exclusive -storage offsite ...`
9. Syncs to the NAS: `duplicacy copy -threads 3 -to offsite` 
10. And finally pings his https://healthchecks.io check, which monitors the backup redundancy.

All used tools and services are free for my personal use-case.
Configuration on new PCs has not been automated (it's only a handful anyway), and the monitoring is binary based on "backed-up or not".
It's the closest to crashplan I could get so far...
I guess the next step would be to build something similar to https://www.duplicati-monitoring.com/

**Comment by TheBestPessimist** [comment]
When discourse is finished, i wanna add this as a greater example of how to duplicacy!

**Comment by zbasbuyuk** [comment]
Thank you gentlemen. I wasn't aware of the scripts and utils. This _*HowToDuplicacy*_  is exactly what I am looking for. Although I still think macro expansion is a good idea,  I can work around. Links shown above are very good examples of free tool usages.

**Comment by opmetal** [comment]
@TheBestPessimist  "@opmetal: is that useful if the other people can go as admin as well? (dk how windows privileges work, but i got all sort of headache from "this file belongs to " for files in my own user profile."

Yes, windows group policy can sit at a higher level than your local Administrator privileges. ie I'm an Administrator on my work pc but certain settings are locked out for me.

**Comment by ZekiBasbuyuk** [comment]
@NiJoao What is 
>2. Simlink-mode to C:/
Thanks.

**Comment by NiJoao** [comment]
@ZekiBasbuyuk I don't know where, and if, it is in the documentation, but read the first response of this issue:
https://duplicacy.com/issue?id=5630721452408832 

> If you create a symbolic link right under the repository directory, Duplicacy will follow the symbolic link to scan the directory it refers to. For symbolic links residing on a subdirectory of the repository, Duplicacy does not follow them and simply backs up them as symbolic links.

It is discussed in multiple threads.

Therefore, the first lines of my Powershell script run as Admin are:

```
mkdir c:\Duplicacy\
cd c:\Duplicacy\
cmd /c 'mklink /D target c:\'

# Copy \.duplicacy\ folder to c:\Duplicacy\, containing .EXEs and \duplicacy utils\

.duplicacy\duplicacy_win_x64_2.1.0.exe init -e -c 4M $env:computername sftp://USER@SERVER/Duplicacy
...
```

This way, the symlink to c:\ is followed and the whole drive is backed up.
You can also add symlinks to all other desirable drives or folders outside c:\
